pkg/abi/abidecode.go

```diff
@@ -94,7 +94,8 @@ func decodeABIElement(ctx context.Context, breadcrumbs string, block []byte, hea
 		}
 		headStart += headOffset
 		headPosition = headStart
-		return walkDynamicChildArrayABIBytes(ctx, "tup", breadcrumbs, block, headOffset, headPosition, component, component.tupleChildren)
+		_, cv, err := walkDynamicChildArrayABIBytes(ctx, "tup", breadcrumbs, block, headStart, headPosition, component, component.tupleChildren)
+		return 32, cv, err
```

Comment:
I was reading your change, and convinced myself of your logic @EnriqueL8 
I wrote an extra test case to try and help myself understand, then I found a related bug on the same line - we were returning the full length of the `walkDynamicChildArrayABIBytes` to the `headBytesRead` of `decodeABIElement` - but actually, only the 32bytes of the `headOffset` were from the header.

The rest (per your change) is consumed from a new header, which starts at `headStart+ headOffset`

Reply:
Makes sense! That is probably why we were seeing issues with nested tuples